feature/search/impl/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/search/impl/navigation/SearchEntryProvider.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.nowinandroid.feature.search.impl.navigation
+
+import androidx.compose.runtime.snapshots.SnapshotStateList
+import androidx.navigation3.runtime.EntryProviderBuilder
+import androidx.navigation3.runtime.entry
+import com.google.samples.apps.nowinandroid.core.navigation.NiaBackStack
+import com.google.samples.apps.nowinandroid.feature.interests.api.navigation.navigateToInterests
+import com.google.samples.apps.nowinandroid.feature.search.api.navigation.SearchRouteNav3
+import com.google.samples.apps.nowinandroid.feature.search.impl.SearchRoute
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.components.ActivityComponent
+import dagger.multibindings.IntoSet
+
+@Module
+@InstallIn(ActivityComponent::class)
+object SearchModule {
+
+    @Provides
+    @IntoSet
+    fun provideSearchEntryProviderBuilder(
+        backStack: NiaBackStack,
+    ): EntryProviderBuilder<Any>.() -> @JvmSuppressWildcards Unit = {
+        entry<SearchRouteNav3> { key ->
+            SearchRoute(
+                onBackClick = backStack::removeLast,
```

Comment:
This is required because `SearchScreen` displays a back arrow (do we need it?). Is it always the case that we just pop the last back stack item when it's clicked? Need to think about this more.

Reply:
I think it's needed. The search screen covers the entire screen so there's really no convenient way to dismiss/go back from search screen without it. Users could use system back, but then this would apply to all back arrows in general - I believe the added convenience from back arrow has its value.

Or in other words, is there any harm to having it? So far the back behavior of removeLost() seems natural to me as I play around with search. Unless you wanted to make search into a list/detail screen?

Reply:
Lets TODO this in https://github.com/android/nowinandroid/issues/1934

Reply:
Agreed that because the SearchScreen has responsibility for the back button it is ok for it to always just remove itself.